### PR TITLE
feature/44_REPLY_FUNCTIONALITY

### PR DESCRIPTION
**Pull request by Symphonie234**
### Changes Made:

- Modified the **`CommentController`** to handle replies.
- Created two new **Requests** files named **`ReplyToCommentRequest.php`** and **`StoreReplyRequest.php`**.
- Modified the **`Comment`** model to support replies.
- Created two new **Model** files named **`Reply.php`** and **`ReplyLike.php`**.
- Added two new **migration** files named **`create_replies_table`** and **`create_reply_likes_table`**.
- Added new **CSS** to the **`listings_show.css`** file.
- Created new **messages** for replies in the **`lang/en/messages.php file`**.
- Adjusted the code in the **`listings/show.blade.php`** file to accommodate replies.
- Created new routes in the **`web.php`** route file.

### CODE REVIEW

File: app/Http/Controllers/CommentController.php [hunk 1]
```diff
@@ -31,26 +34,47 @@ public function __construct()
...
     public function store(CommentStoreRequest $request)
     {
-        // Use the register method from the trait
-        $comment = Comment::register(
-            [
+        // Get the parent_id from the form data
+        $parentId = $request->input('parent_id');
+
+        // Check if a parent_id is provided
+        if ($parentId) {
```
Comment (ChronoDevs) [inline app/Http/Controllers/CommentController.php]: Please transfer this process in the Comment Model

File: app/Http/Controllers/CommentController.php [hunk 2]
```diff
@@ -144,4 +172,113 @@ public function unlike($commentId)
...
+     * @param  int  $commentId
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function reply(ReplyToCommentRequest $request, $commentId)
+    {
+        // Find the parent comment
+        $parentComment = Comment::findOrFail($commentId);
+
+        // Create a new reply comment
+        $comment = new Comment([
```
Comment (ChronoDevs) [inline app/Http/Controllers/CommentController.php]: use the function you created in the Comment Model Register or Register trait

File: app/Http/Controllers/CommentController.php [hunk 3]
```diff
@@ -144,4 +172,113 @@ public function unlike($commentId)
...
+     * @param  StoreReplyToCommentRequest  $request
+     * @param  int  $commentId
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function storeReply(StoreReplyRequest $request, $commentId)
+    {
+        $comment = Comment::findOrFail($commentId);
+
+        // Create a new reply with the content and associate it with the authenticated user
+        $reply = new Reply([
```
Comment (ChronoDevs) [inline app/Http/Controllers/CommentController.php]: make error response if failed and this process should be in the model

File: app/Http/Controllers/CommentController.php [hunk 4]
```diff
@@ -144,4 +172,113 @@ public function unlike($commentId)
...
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function likeReply($replyId)
+    {
+        $reply = Reply::findOrFail($replyId);
+
+        // Check if the user has already liked the reply
+        $userId = auth()->user()->id;
+        if (!$reply->isLikedByUser($userId)) {
+            $like = new ReplyLike([
```
Comment (ChronoDevs) [inline app/Http/Controllers/CommentController.php]: transfer this process in the model

**Comment by Symphonie234** [comment]
### Changes Addressed in PR No. 1:

- Transferred the process of storing comments to the **`Comment`** **Model**.
- Utilized the **`Register`** **Trait** and transferred the contents of the reply function to the **`Comment`** **Model**.
- Implemented an error response if the reply function fails at the **`CommentController.php`**.
- Transferred the processes in the **like** and **unlike** functions to the **`Reply`** **Model**.

**Pull request merged**